embrace-android-delivery/src/test/kotlin/io/embrace/android/embracesdk/internal/delivery/execution/HttpUrlConnectionRequestExecutionServiceTest.kt

```diff
@@ -0,0 +1,204 @@
+package io.embrace.android.embracesdk.internal.delivery.execution
+
+import io.embrace.android.embracesdk.internal.comms.api.ApiResponse
+import io.embrace.android.embracesdk.internal.delivery.SupportedEnvelopeType
+import okhttp3.Headers.Companion.toHeaders
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import okhttp3.mockwebserver.SocketPolicy
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import java.net.UnknownHostException
+
+class HttpUrlConnectionRequestExecutionServiceTest {
```

Comment:
In theory since the APIs are the same, this test class can be used to test both implementations provided you reference the interface and not the payload, right?